flang/tools/bbc/bbc.cpp

```diff
@@ -217,22 +210,29 @@ static void convertFortranSourceToMLIR(
   llvm::raw_fd_ostream out(outputName, ec);
   if (ec) {
     llvm::errs() << "could not open output file " << outputName << '\n';
-    return;
+    return mlir::failure();
   }
 
   // Otherwise run the default passes.
   mlir::PassManager pm(mlirModule.getContext());
   mlir::applyPassManagerCLOptions(pm);
   if (passPipeline.hasAnyOccurrences()) {
+    // run the command-line specified pipeline
     passPipeline.addToPipeline(pm);
   } else if (emitFIR) {
-    // --emit-fir: Build the IR, verify it, and dump the IR (unconditionally).
+    // --emit-fir: Build the IR, verify it, and dump the IR if the IR passes
+    // verification. Use --dump-module-on-failure to dump invalid IR.
```

Comment:
FYI - I do find both of these options to be useful.

Reply:
Me too. It's a little easier if the error message and the module output can be selectively enabled.